src/api/tokenList/Subscriptions.ts

```diff
@@ -0,0 +1,32 @@
+import { Command } from 'types'
+
+type SubscriptionCallback<T> = (newState: T) => void
+
+export interface SubscriptionsInterface<State> {
+  subscribe(callback: SubscriptionCallback<State>): Command
+  unsubscribe(callback: SubscriptionCallback<State>): void
+  unsubscribeAll(): void
+  triggerSubscriptions(newState: State): void
+}
+
+export default class GenericSubscriptions<State> implements SubscriptionsInterface<State> {
```

Comment:
One idea. Why don't you create a new interface `Subscriptions<State>` and then do `GenericSubscriptions` implement this interface plus add the public `triggerSubscriptions` method.

This way TokenList would extend Subscriptions, and the Impl/Mock would extend GenericSubscriptions

Reply:
Can't extend interface
Can only extend class `GenericSubscriptions`, so `triggerSubscriptions` becomes public, so we remove it from interface with 
```ts
export interface TokenList extends Omit<SubscriptionsInterface<TokenDetails[]>, 'triggerSubscriptions'>
```

But you gave me an idea, I'll make it `protected`, 6abef37b3728105c8fe616875be650b913c81729